src/main/java/com/onarandombox/MultiversePortals/PortalPlayerSession.java

```diff
@@ -282,31 +282,55 @@ public void setTeleportTime(Date date) {
         this.lastTeleportTime = date;
     }
 
-    public boolean allowTeleportViaCooldown(Date date) {
-        return (date.after(new Date(this.lastTeleportTime.getTime() + this.plugin.getCooldownTime())));
+    /**
+     * Get the remaining teleport cooldown time in milliseconds.
+     * If the cooldown is finished, 0 will be returned.
+     *
+     * @return The remaining teleport cooldown time in milliseconds, or 0 if the cooldown is over.
+     */
+    private long getRemainingTeleportCooldown(Date time) {
```

Comment:
I think the time parameter is not needed? Bcu it should always be the current time and not something passed by when calling the method which could potentially be not current time. So you can just created the Date object in the method.

Reply:
Yeah, good point.